src/connection/mod.rs

```diff
@@ -139,4 +139,8 @@ pub trait IpmiCommand: Into<Message> {
             Err(ParseResponseError::Failed(cc))
         }
     }
+
+    fn address_and_channel(&self) -> Option<(u8, u8)> {
```

Comment:
I think we should have [newtypes] for `IpmbChannel` and `IpmbAddress` (or whatever naming is correct, not 100% sure which it is these are referring to). It will make it a bit clearer what you're passing around, and is a nice zero-cost abstraction in this case! 

Mostly adding this comment because I would prefer we have hard-typed values as far as possible, since the amount of `u8`'s, `i16`'s and what have you could get confusing quickly, otherwise :)

[newtypes]: https://doc.rust-lang.org/rust-by-example/generics/new_types.html

Reply:
Done